suite/tests/client-interface/avx512ctx.c

```diff
@@ -0,0 +1,306 @@
+        vmovups  zmm23, [REG_XCX + 0x5c0]
+        vmovups  zmm24, [REG_XCX + 0x600]
+        vmovups  zmm25, [REG_XCX + 0x640]
+        vmovups  zmm26, [REG_XCX + 0x680]
+        vmovups  zmm27, [REG_XCX + 0x6c0]
+        vmovups  zmm28, [REG_XCX + 0x700]
+        vmovups  zmm29, [REG_XCX + 0x740]
+        vmovups  zmm30, [REG_XCX + 0x780]
+        vmovups  zmm31, [REG_XCX + 0x7c0]
+#endif
+
+        add      REG_XSP, FRAME_PADDING
+        POP_CALLEE_SAVED_REGS()
+        ret
+        END_FUNC(FUNCNAME)
+#undef FUNCNAME
+
+#define FUNCNAME get_zmm
+        DECLARE_FUNC_SEH(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+        mov      REG_XCX, ARG1
+        PUSH_CALLEE_SAVED_REGS()
+        sub      REG_XSP, FRAME_PADDING
+        END_PROLOG
+
+       vmovups  [REG_XCX],         zmm0
+       vmovups  [REG_XCX + 0x40],  zmm1
+       vmovups  [REG_XCX + 0x80],  zmm2
+       vmovups  [REG_XCX + 0xc0],  zmm3
+       vmovups  [REG_XCX + 0x100], zmm4
+       vmovups  [REG_XCX + 0x140], zmm5
+       vmovups  [REG_XCX + 0x180], zmm6
+       vmovups  [REG_XCX + 0x1c0], zmm7
+#ifdef X64
+       vmovups  [REG_XCX + 0x200], zmm8
+       vmovups  [REG_XCX + 0x240], zmm9
+       vmovups  [REG_XCX + 0x280], zmm10
+       vmovups  [REG_XCX + 0x2c0], zmm11
+       vmovups  [REG_XCX + 0x300], zmm12
+       vmovups  [REG_XCX + 0x340], zmm13
+       vmovups  [REG_XCX + 0x380], zmm14
+       vmovups  [REG_XCX + 0x3c0], zmm15
+       vmovups  [REG_XCX + 0x400], zmm16
+       vmovups  [REG_XCX + 0x440], zmm17
+       vmovups  [REG_XCX + 0x480], zmm18
+       vmovups  [REG_XCX + 0x4c0], zmm19
+       vmovups  [REG_XCX + 0x500], zmm20
+       vmovups  [REG_XCX + 0x540], zmm12
+       vmovups  [REG_XCX + 0x580], zmm22
+       vmovups  [REG_XCX + 0x5c0], zmm23
+       vmovups  [REG_XCX + 0x600], zmm24
+       vmovups  [REG_XCX + 0x640], zmm25
+       vmovups  [REG_XCX + 0x680], zmm26
+       vmovups  [REG_XCX + 0x6c0], zmm27
+       vmovups  [REG_XCX + 0x700], zmm28
+       vmovups  [REG_XCX + 0x740], zmm29
+       vmovups  [REG_XCX + 0x780], zmm30
+       vmovups  [REG_XCX + 0x7c0], zmm31
+#endif
+
+        add      REG_XSP, FRAME_PADDING
+        POP_CALLEE_SAVED_REGS()
+        ret
+        END_FUNC(FUNCNAME)
+#undef FUNCNAME
+
+#define FUNCNAME init_opmask
+        DECLARE_FUNC_SEH(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+        mov      REG_XCX, ARG1
+        PUSH_CALLEE_SAVED_REGS()
+        sub      REG_XSP, FRAME_PADDING
+        END_PROLOG
+
+        mov      ax, WORD [REG_XCX]
+        kmovw    k0, eax
+        mov      ax, WORD [REG_XCX + 0x2]
+        kmovw    k1, eax
+        mov      ax, WORD [REG_XCX + 0x4]
+        kmovw    k2, eax
+        mov      ax, WORD [REG_XCX + 0x6]
+        kmovw    k3, eax
+        mov      ax, WORD [REG_XCX + 0x8]
+        kmovw    k4, eax
+        mov      ax, WORD [REG_XCX + 0xa]
+        kmovw    k5, eax
+        mov      ax, WORD [REG_XCX + 0xc]
+        kmovw    k6, eax
+        mov      ax, WORD [REG_XCX + 0xe]
```

Comment:
Same thing: explicit multiplication is often easier to read

Reply:
Done.